controllers/cloudinit/controlplane_init.go

```diff
@@ -141,7 +143,7 @@ func NewInitControlPlane(input *ControlPlaneInitInput) (*CloudConfig, error) {
 	cloudConfig.RunCommands = append(cloudConfig.RunCommands, input.PreRunCommands...)
 	cloudConfig.RunCommands = append(cloudConfig.RunCommands,
 		fmt.Sprintf("%s %q %q", scriptPath(snapstoreHTTPProxyScript), input.SnapstoreHTTPProxy, input.SnapstoreHTTPSProxy),
-		fmt.Sprintf("%s %q %q", scriptPath(snapstoreProxyScript), input.SnapstoreProxyDomain, input.SnapstoreProxyId),
+		fmt.Sprintf("%s %q %q %q", scriptPath(snapstoreProxyScript), input.SnapstoreProxyScheme, input.SnapstoreProxyDomain, input.SnapstoreProxyId),
```

Comment:
This will always be unset, you need to update the controller to pass this field from API -> ControlPlaneFooInput and WorkerInput, e.g. https://github.com/canonical/cluster-api-bootstrap-provider-microk8s/blob/18e470f9cb02a32e90be99a1a4c9b28ffc984e35/controllers/microk8sconfig_controller.go#L403-L427